src/Sulu/Bundle/MediaBundle/Media/Storage/LocalStorage.php

```diff
@@ -144,21 +155,41 @@ private function getUniqueFileName(string $folder, string $fileName, int $counte
 
         ++$counter;
 
-        return $this->getUniqueFileName($folder, $fileName, $counter);
+        return $this->getUniqueFileName($parentPath, $fileName, $counter);
     }
 
-    private function getPathByFolderAndFileName(string $folder, string $fileName): string
+    private function getFilesystemPath(?string $directory, ?string $segment = null, ?string $fileName = null): string
     {
-        return \rtrim($folder, '/') . '/' . \ltrim($fileName, '/');
+        return \implode('/', \array_filter([$this->uploadPath, $directory, $segment, $fileName]));
     }
 
-    private function createPath(string $segment, ?string $fileName = null): string
+    /**
+     * @param array<string, string|null> $storageOptions
+     */
+    private function getStorageOption(array $storageOptions, string $key): ?string
     {
-        return \implode('/', \array_filter([$this->uploadPath, $segment, $fileName]));
+        return \array_key_exists($key, $storageOptions) ? $storageOptions[$key] : null;
     }
 
-    private function getStorageOption(array $storageOption, string $key): ?string
+    /**
+     * @param array<string, string|null> $storageOptions
+     */
+    private function createDirectories(array $storageOptions): void
```

Comment:
This method is unnecessary the previous  `$segmentPath` would just need to keep the `directory` parameter in count. 

https://github.com/sulu/sulu/blob/02161cb45e0c7175c267ceaea3ed310a32c996fe/src/Sulu/Bundle/MediaBundle/Media/Storage/LocalStorage.php#L61

Then already at:

https://github.com/sulu/sulu/blob/02161cb45e0c7175c267ceaea3ed310a32c996fe/src/Sulu/Bundle/MediaBundle/Media/Storage/LocalStorage.php#L69

The correct directory would be created recursively by the symfony filesystem component: https://symfony.com/doc/current/components/filesystem.html#mkdir

Reply:
okay i have removed the check for the `directoryPath`. think i would keep the private method because we need to do this in the `save` method and the `move` method